project/yxuan0329/proposal.rst

```diff
@@ -0,0 +1,186 @@
+=================
+R-Tree Spatial Search Engine
+=================
+
+
+..   You are encouraged to use this plain-text `reStructuredText
+..   <https://docutils.sourceforge.io/rst.html>`__ format.  :download:`Download
+..   this file. <project_template.rst>`
+
+.. Please do give your project a name that clearly states the subject and is short
+.. enough for people to remember.
+
+.. Additional documents or files may be used.  They should be added in the
+.. repository.
+
+Basic Information
+=================
+
+.. Create a GitHub repository to host your project and add the URL here.  The
+.. GitHub repository has an 'About' field, in which you are encouraged to write a
+.. simple statement (preferably one sentence) to introduce the project.
+
+R-tree is a tree structure designed for efficient spatial data access. It has 
+been widely used in real-world geographic searching because of its exceptional 
+searching performance. In this project, we aim to serve a R-tree library to 
+provide an efficient search for end-users. 
+
+
+Problem to Solve
+================
+
+.. Describe the problem or the set of problems you want to solve.  Include
+.. necessary background information without making it lengthy.
+
+.. Some points may help you organize the problem description:
+
+.. 1. The field or industry of the problem.
+.. 2. The physics and/or the mathematics behind the problem.
+.. 3. The algorithm or numerical method that should be applied for solving the
+..    problem.
+
+In geographic searching application, the users would expect to swiftly receive 
```

Comment:
How do you plan to get test data (for geographical applications)?

Reply:
I'm planning to write a Python script to generate the test data randomly. For the initial testing, I will generate the data in 2 or 3 dimensions. After that, I intend to expand the dimensionality of the generated data to 4 or 5 dimensions. This expansion is aimed at testing the capacity to support higher-dimensional data scenarios.

Reply:
It is worth the time to think more about how to devise the script for data generation.  You will need it anyway, so as a plan at this point, it is OK.